File: Data Structures & Algorithms/clone-graph/submission-0.java
/*
Definition for a Node.
class Node {
    public int val;
    public List<Node> neighbors;
    public Node() {
        val = 0;
        neighbors = new ArrayList<Node>();
    }
    public Node(int _val) {
        val = _val;
        neighbors = new ArrayList<Node>();
    }
    public Node(int _val, ArrayList<Node> _neighbors) {
        val = _val;
        neighbors = _neighbors;
    }
}
*/

class Solution {
    private Node solve(Node node, Map<Node, Node> mp) {
        if(node == null || mp.containsKey(node)) return mp.get(node);
        Node newNode = new Node(node.val);
        mp.put(node, newNode);
        for(var nbr : node.neighbors) {
            newNode.neighbors.add(solve(nbr, mp));
        }
        return newNode;
    }
    public Node cloneGraph(Node node) {
        Map<Node, Node> mp = new HashMap<>();
        return solve(node, mp);
        // return mp.get(node);
    }
}

File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    int solve(int n) {
        int ans=0;
        while(n) {
            ans += n&1;
            n >>= 1;
        }
        return ans;
    }
    vector<int> countBits(int n) {
        vector<int> ans(n+1, 0);
        for(int i=0; i<=n; i++) ans[i] = solve(i);
        return ans;;
    }
};
